### ISSUE TITLE
Bug: `@types/node` aliased types don't match `typeOrValueSpecifier`

### ISSUE THREAD

**Issue by kirkwaiblinger** [typescript-eslint/typescript-eslint#11504]
### Before You File a Bug Report Please Confirm You Have Done The Following...

- [x] I have tried restarting my IDE and the issue persists.
- [x] I have updated to the latest version of the packages.
- [x] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3Abug) and found none that matched my issue.
- [x] I have [read the FAQ](https://typescript-eslint.io/linting/troubleshooting) and my problem is not listed.

### Relevant Package

ast-spec

### Playground Link

https://typescript-eslint.io/play/#ts=5.7.2&showAST=es&fileType=.tsx&code=PTAEBcFMGdwOnNAUASwLYAcD2AncoBvUAExgGMcUAjSUAX1ADMcs1QByAOy1IC4pY7ANxIkpaBWqQAFAEohoEKEB8G4BO90EA&eslintrc=N4KABGBEBOCuA2BTAzpAXGUEKQAIBcBPABxQGNoBLY-AWhXkoDt8B6Jge1oDN4OBDfMwDmtYtA4BbSshTowAbXDYcAd37QmkADRhlK4HpUrI-eH1UAxDtADSnVUwDK-bogDCZ%2BKgxLjxrH9jSG4JSXlIYn4yAGt%2BYUQdfSCcJn5JRIxIABNyKgAjRO1klMjouISIzly0fBR8SBLjAF8miABdJtaVTohW5qA&tsconfig=N4KABGBEDGD2C2AHAlgGwKYCcDyiAuysAdgM6QBcYoEEkJemy0eAcgK6qoDCAFutAGsylBm3TgwAXxCSgA&tokens=false

### Repro Code

```TypeScript
// test.ts
import { describe } from 'node:test';

describe(); // 🫤
```

### ESLint Config

```javascript
{
  "rules": {
    "@typescript-eslint/no-floating-promises": [
      "warn", 
      { 
        "allowForKnownSafeCalls": [
          {
            "from": "package",
            "name": "describe",
            "package": "node:test"
          }
        ]
      }
    ]
  }
}
```

### tsconfig

```jsonc
{
  "compilerOptions": {
    // ...
  }
}
```

### Expected Result

`describe()` should not report, due to matching with `typeOrValueSpecifier`

### Actual Result

report on `describe()` 

### Additional Info

Followup to https://github.com/typescript-eslint/typescript-eslint/issues/11372#issuecomment-3075084147

cc @Renegade334 

### Versions

| package                                | version |
| -------------------------------------- | ------- |
| `typescript-eslint`     | `8.40.0` |
| `@types/node`            | `22.17.12` |

**Comment by Renegade334** [comment]
Sorry, I evidently forgot to spin off a new issue, many thanks!

I haven't looked too closely, but does #10670 unlock matching by aliased symbol?

**Comment by kirkwaiblinger** [comment]
> Sorry, I evidently forgot to spin off a new issue, many thanks!

No problem! I just needed to have a link since CI failed after upgrading `@types/node`, hehe. See https://github.com/typescript-eslint/typescript-eslint/pull/11505.

> I haven't looked too closely, but does [#10670](https://github.com/typescript-eslint/typescript-eslint/pull/10670) unlock matching by aliased symbol?

🤷 Only one way to find out! 

Though, rereading https://github.com/typescript-eslint/typescript-eslint/issues/11372#issuecomment-3075084147, I'm a little worried that this will be an unfixable duplicate of https://github.com/typescript-eslint/typescript-eslint/issues/10053 😬.

**Comment by JoshuaKGoldberg** [comment]
😬. We'll definitely need to support this, I'm afraid. It's just such a common need - basically one of the main motivators behind TypeOrValueSpecifier!

**Comment by bijela-gora** [comment]
@JoshuaKGoldberg hi! Looks like this issue was fixed.

Is this possible to support this?

<img width="365" height="164" alt="Image" src="https://github.com/user-attachments/assets/3cdf54d1-8a1d-4fa1-8ed7-4a8c75935aaf" />

<img width="468" height="340" alt="Image" src="https://github.com/user-attachments/assets/213abb2c-24d3-470d-a783-b7045f36bc0c" />

Playground: https://typescript-eslint.io/play/#ts=5.7.2&fileType=.tsx&code=PTAEBcFMGdwOnNAUASwLYAcD2AncoBvUAExgGMcUAjSAGlBXwF9QAzHLNUAcgDstSALiixuAbiRJS0CtUgAKbnDjd68gJSgAvAD5CSUA3CLlKtZt37DhkAdBN1dxnGgBrFBhPLVoDdr1EDqAgoIB8G4Ane6BIDmJAA&eslintrc=N4KABGBEBOCuA2BTAzpAXGUEKQAIBcBPABxQGNoBLY-AWhXkoDt8B6Jge1oDN4OBDfMwDmtYtA4BbSshTowAbXDYcAd37QmkADRhlK4HpUrI-eH1UAxDtADSnVUwDK-bogDCZ%2BKgxLjxrH9jSG4JSXlIYn4yAGt%2BYUQdfSCcJn5JRN9IABNyKgAjRN1ISnwdKFKAOmQY6kgAXW1klMjouISIzly0fBQy5uMAXwGIeoHhlTGIYcGgA&tsconfig=N4KABGBEDGD2C2AHAlgGwKYCcDyiAuysAdgM6QBcYoEEkJemy0eAcgK6qoDCAFutAGsylBm3TgwAXxCSgA&tokens=false

**Comment by JoshuaKGoldberg** [comment]
> Looks like this issue was fixed.

Nice, it was! The original repro case and that new one's first part both work. Lovely.

> Is this possible to support this?
> (playground showing `it.skip` still being reported)

Hopefully. Tracked in: #10740. It's tricky, but if someone can figure it out that'd be lovely.

Cheers!

**Comment by Renegade334** [comment]
> > Looks like this issue was fixed.
> 
> Nice, it was! The original repro case and that new one's first part both work. Lovely.

A byproduct of #11430, I assume. Nice one! 🥳
